src/resources/views/crud/fields/relationship/fetch.blade.php

```diff
@@ -336,20 +336,38 @@ function bpFieldInitFetchElement(element) {
         if (!$(element).hasClass("select2-hidden-accessible"))
         {
             $(element).select2($select2Settings);
-             // if any dependencies have been declared
+
+            // if any dependencies have been declared
             // when one of those dependencies changes value
             // reset the select2 value
             for (var i=0; i < $dependencies.length; i++) {
-                $dependency = $dependencies[i];
-                $('input[name='+$dependency+'], select[name='+$dependency+'], checkbox[name='+$dependency+'], radio[name='+$dependency+'], textarea[name='+$dependency+']').change(function () {
-
-                //apart from setting selection to null, we clear the options until the next fetch from server happen.
-                $(element.find('option:not([value=""])')).remove();
-
-                element.val(null).trigger("change");
-
-                });
+                var $dependency = $dependencies[i];
+                //if element does not have a custom-selector attribute we use the name attribute
+                if(typeof element.attr('data-custom-selector') == 'undefined') {
+                    form.find(`[name="${$dependency}"], [name="${$dependency}[]"]`).change(function(el) {
+                            $(element.find('option:not([value=""])')).remove();
+                            element.val(null).trigger("change");
+                    });
+                }else{
+                    // we get the row number and custom selector from where element is called
+                    let rowNumber = element.attr('data-row-number');
+                    let selector = element.attr('data-custom-selector');
+
+                    // replace in the custom selector string the corresponding row and dependency name to match
+                    selector = selector.replace(/%\w+%/g, function(replace) {
+                        if(replace == "%DEPENDENCY%") {
+                            return $dependency;
+                        }
+                        if(replace == "%ROW%") {
+                            return rowNumber;
+                        }
+                    });
```

Comment:
```suggestion
                    selector = selector
                        .replaceAll('%DEPENDENCY%', $dependency)
                        .replaceAll('%ROW%', rowNumber);
```

@pxpm is this enough?